### PR TITLE
CMDEV-26 sample code for amazon-sqs-java-messaging-lib

### PR DESCRIPTION
**Pull request by msfuko**
* Sync send / Sync Recv / Async receive with request-reply message
* Below to-do will not keep going while we have another sample template
  - unit testing
  - Spring annotation (I've tried some...but somehow did not work well :-S)

### CODE REVIEW

**Comment by msfuko** [comment]
Trigger url-
Send sample message - localhost:8080/spring-jms-sample/rest/message/jms-sqs/test/send
Sync receive message - localhost:8080/spring-jms-sample/rest/message/jms-sqs/test/sync-receive
Async receive message listener setup - localhost:8080/spring-jms-sample/rest/message/jms-sqs/test/async-receive

**Comment by msfuko** [comment]
Hi @hidetosaito ,

Please kindly also take a look for the uncompleted code :-P

It's uglier though but with a little different perspective with your baby.
Although it won't be continued implementing, but it would be very appreciated if you have any comments and suggestions for it :-)

Thank you very much.

Regards,
Chloe

File: spring-jms-sample/src/main/java/com/trendmicro/dcs/springjmssample/api/utils/messaging/processor/SampleMessageConsumer.java [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+	private String replyQueueName;
+	
+	private Log logger = LogFactory.getLog(this.getClass());
+
+	public void checkReplyMessage(Message message) throws JMSException {
+		Session session = connection.createSession(false, Session.AUTO_ACKNOWLEDGE);
+		/*
+		 * FIXME: I could not make it work by getting JMSReplyTo header from message.
+		 *        It is always null although I've set it in MessageProducer 
+		 *        Not sure it's supported or not :-(
```
Comment (hidetosaito) [inline spring-jms-sample/src/main/java/com/trendmicro/dcs/springjmssample/api/utils/messaging/processor/SampleMessageConsumer.java]: probably not supported :-(
you can verify at AWS Console to take a look detail entity data.

sqs1

sqs2

for your understanding, i recommend to play with ActiveMQ or other JMS compliant message queue server to verify your code.

File: spring-jms-sample/src/main/java/com/trendmicro/dcs/springjmssample/api/utils/messaging/processor/SampleMessageConsumer.java [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+	@Resource(name = "invalidQueueName")
+	private String invalidQueueName;
+
+	@Resource(name = "replyQueueName")
+	private String replyQueueName;
+	
+	private Log logger = LogFactory.getLog(this.getClass());
+
+	public void checkReplyMessage(Message message) throws JMSException {
+		Session session = connection.createSession(false, Session.AUTO_ACKNOWLEDGE);
```
Comment (hidetosaito) [inline spring-jms-sample/src/main/java/com/trendmicro/dcs/springjmssample/api/utils/messaging/processor/SampleMessageConsumer.java]: I think, we need to care about a connection pooling and transaction as similar as database. Spring should have it, please keep in mind and research for the next step.

**Comment by hidetosaito** [comment]
Very good start and your are quick learner ! and we are almost writing the similar code at the same time :-)

The next steps are

1. understand full functionality of JMS, i recommend to play with full stack JMS server such as ActiveMQ or JBOSS then compare with SQS. SQS probably lacks some of JMS spec. so we should avoid this limitation
2. Use SpringFramework to abstracting queue client implementation, in another word, shouldn't import *com.amazon.sqs.javamessaging* in service layer. as similar as Omnibus database implementation, please refer to my sample code
3. i'd prefer to use in memory queue server for unit testing instead of SQS directly, because we should test the logic itself right?, and not easy to embed AWS access/secret key in the code. my sample code uses ActiveMQ in memory mode, but it seems it raises RuntimeException when finish the unit test. perhaps need to care for graceful shutdown
4. please follow my request of https://github.com/TrendMicroDCS/prototype-sample/pull/3
5. in my sample code, i adopt DAO design pattern, not sure it is good for queue architecture or not, please also keep in mind to pursuit the better design pattern
6. (optional) compare with Spring Integration whether we can reduce the code or not

**Pull request merged**